Modules/selectmodule.c

```diff
@@ -1427,77 +1429,79 @@ pyepoll_internal_ctl(int epfd, int op, PyObject *pfd, unsigned int events)
 {
-    PyObject *pfd;
-    unsigned int events = EPOLLIN | EPOLLOUT | EPOLLPRI;
-    static char *kwlist[] = {"fd", "eventmask", NULL};
+    return pyepoll_internal_ctl(self->epfd, EPOLL_CTL_ADD, fd, eventmask);
+}
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|I:register", kwlist,
-                                     &pfd, &events)) {
-        return NULL;
-    }
+/*[clinic input]
+select.epoll.modify
 
-    return pyepoll_internal_ctl(self->epfd, EPOLL_CTL_ADD, pfd, events);
-}
+    fd: fildes
+      the target file descriptor of the operation
+    eventmask: unsigned_int(bitwise=True)
+      a bit set composed of the various EPOLL constants
 
-PyDoc_STRVAR(pyepoll_register_doc,
-"register(fd[, eventmask]) -> None\n\
-\n\
-Registers a new fd or raises an OSError if the fd is already registered.\n\
-fd is the target file descriptor of the operation.\n\
-events is a bit set composed of the various EPOLL constants; the default\n\
-is EPOLLIN | EPOLLOUT | EPOLLPRI.\n\
-\n\
-The epoll interface supports all file descriptors that support poll.");
+Modify event mask for a registered file descriptor.
+[clinic start generated code]*/
 
 static PyObject *
-pyepoll_modify(pyEpoll_Object *self, PyObject *args, PyObject *kwds)
+select_epoll_modify_impl(pyEpoll_Object *self, int fd,
+                         unsigned int eventmask)
+/*[clinic end generated code: output=7e3447307cff6f65 input=88a83dac53a8c3da]*/
 {
-    PyObject *pfd;
-    unsigned int events;
-    static char *kwlist[] = {"fd", "eventmask", NULL};
+    return pyepoll_internal_ctl(self->epfd, EPOLL_CTL_MOD, fd, eventmask);
+}
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OI:modify", kwlist,
-                                     &pfd, &events)) {
-        return NULL;
-    }
+/*[clinic input]
+select.epoll.unregister
 
-    return pyepoll_internal_ctl(self->epfd, EPOLL_CTL_MOD, pfd, events);
-}
+    fd: fildes
+      the target file descriptor of the operation
 
-PyDoc_STRVAR(pyepoll_modify_doc,
-"modify(fd, eventmask) -> None\n\
-\n\
-fd is the target file descriptor of the operation\n\
-events is a bit set composed of the various EPOLL constants");
+Remove a registered file descriptor from the epoll object.
+[clinic start generated code]*/
 
 static PyObject *
-pyepoll_unregister(pyEpoll_Object *self, PyObject *args, PyObject *kwds)
+select_epoll_unregister_impl(pyEpoll_Object *self, int fd)
+/*[clinic end generated code: output=07c5dbd612a512d4 input=3093f68d3644743d]*/
 {
-    PyObject *pfd;
-    static char *kwlist[] = {"fd", NULL};
+    return pyepoll_internal_ctl(self->epfd, EPOLL_CTL_DEL, fd, 0);
+}
 
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O:unregister", kwlist,
-                                     &pfd)) {
-        return NULL;
-    }
+/*[clinic input]
+select.epoll.poll
 
-    return pyepoll_internal_ctl(self->epfd, EPOLL_CTL_DEL, pfd, 0);
-}
+    timeout as timeout_obj: object(c_default="Py_None") = -1.0
```

Comment:
Why not use None as a default?

Reply:
The doc-string mentions, regarding timeout, "-1 makes poll wait indefinitely" but doesn't mention `None`, though `None` actually does the same. Given this, I thought documenting the default as `-1` was clearer.

Perhaps we should improve the doc-string, mentioning that either `None` or `-1` mean "wait indefinitely", and then change the default to `None`?

Reply:
Do what looks better to you.

Reply:
I'll improve the doc-string and change the default to `None`.